Sources/Common/Core/ClassHierarchy/index.js

```diff
@@ -1,7 +1,14 @@
+/* eslint-disable prefer-rest-params */
 export default class ClassHierarchy extends Array {
-  push(...args) {
-    // no perf issue since args.length should be small
-    const newArgs = args.filter((arg) => !this.includes(arg));
+  push() {
+    // using a for loop because a filter would trigger the garbage collector
+    const newArgs = [];
+    for (let i = 0; i < arguments.length; i++) {
+      if (!this.includes(arguments[i])) {
+        newArgs.push(arguments[i]);
```

Comment:
why don't you call `super.push(arguments[i])` instead of creating a temp `newArgs`?

Reply:
I didn't measure any performance issue with that part of code. I'm also not sure if multiple push will result in better performance.

Reply:
`.push()` is almost always called with 1 argument each time.

Reply:
Initially I was okay with this since `.push()` does have a return value (which is the new length of the array), but now that @finetjul brings this up it would be better to actually call `super.push(arguments[i])` as mentioned and avoid the extraneous newArgs allocation, ending with a `return this.length`.

Reply:
and also it is more correct: it will remove duplicates if they exist in `arguments`